subcommand/server-acl-init/command_test.go

```diff
@@ -0,0 +1,984 @@
+		clientset: k8s,
+	}
+	cmd.init()
+	responseCode := cmd.Run([]string{
+		"-release-name=" + releaseName,
+		"-k8s-namespace=" + ns,
+		"-expected-replicas=1",
+	})
+	require.Equal(0, responseCode, ui.ErrorWriter.String())
+
+	// Test that the bootstrap kube secret is created.
+	bootstrapSecret, err := k8s.CoreV1().Secrets(ns).Get(fmt.Sprintf("%s-consul-bootstrap-acl-token", releaseName), metav1.GetOptions{})
+	require.NoError(err)
+	require.NotNil(bootstrapSecret)
+	token, ok := bootstrapSecret.StringData["token"]
+	require.True(ok)
+
+	// Check that it has the right policies.
+	consul := testAgent.Client()
+	tokenData, _, err := consul.ACL().TokenReadSelf(&api.QueryOptions{Token: token})
+	require.NoError(err)
+	require.Equal("global-management", tokenData.Policies[0].Name)
+
+	// Check that the agent policy was created.
+	policies, _, err := consul.ACL().PolicyList(&api.QueryOptions{Token: token})
+	require.NoError(err)
+	found := false
+	for _, p := range policies {
+		if p.Name == "agent-token" {
+			found = true
+			break
+		}
+	}
+	require.True(found, "agent-token policy was not found")
+
+	// We should also test that the server's token was updated, however I
+	// couldn't find a way to test that with the test agent. Instead we test
+	// that in another test when we're using an httptest server instead of
+	// the test agent and we can assert that the /v1/agent/token/agent
+	// endpoint was called.
+}
+
+// Test the different flags that should create tokens and save them as
+// Kubernetes secrets.
+func TestRun_Tokens(t *testing.T) {
+	t.Parallel()
+
+	cases := map[string]struct {
+		Flag      string
+		TokenName string
+	}{
+		"client token": {
+			"-create-client-token",
+			"client",
+		},
+		"catalog-sync token": {
+			"-create-sync-token",
+			"catalog-sync",
+		},
+		"enterprise-license token": {
+			"-create-enterprise-license-token",
+			"enterprise-license",
+		},
+		"snapshot-agent token": {
+			"-create-snapshot-agent-token",
+			"client-snapshot-agent",
+		},
+		"mesh-gateway token": {
+			"-create-mesh-gateway-token",
+			"mesh-gateway",
+		},
+	}
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			k8s, testAgent := completeSetup(t)
+			defer testAgent.Shutdown()
+			require := require.New(t)
+
+			// Run the command.
+			ui := cli.NewMockUi()
+			cmd := Command{
+				UI:        ui,
+				clientset: k8s,
+			}
+			cmd.init()
+			responseCode := cmd.Run([]string{
+				"-release-name=" + releaseName,
+				"-k8s-namespace=" + ns,
+				"-expected-replicas=1",
+				c.Flag,
+			})
+			require.Equal(0, responseCode, ui.ErrorWriter.String())
+
+			// Get the bootstrap token from the kube secret so we can make API calls
+			// to Consul.
+			bootstrapSecret, err := k8s.CoreV1().Secrets(ns).Get(fmt.Sprintf("%s-consul-bootstrap-acl-token", releaseName), metav1.GetOptions{})
+			require.NoError(err)
+			require.NotNil(bootstrapSecret)
+			bootToken, ok := bootstrapSecret.StringData["token"]
+			require.True(ok)
```

Comment:
I noticed that this gets repeated a few times. Although I generally don't think that tests should be DRY, it might be worth extracting it into a helper.

Reply:
Good call, updated.